woocommerce/woocommerce-ios
# In-app purchases debug screen

Closes: #7833
To be merged after #7851 


### Description


In #7832, we added a feature flag to enable IAP in development builds. However, we still don't have the right place to hook this up in site creation, so this adds a new entry in the Hub menu where we can add a UI to test the IAP features.

Because this is more of a developer tool than an in-progress final feature, I'm hiding this behind a beta feature so it's disabled by default.

### Testing instructions


1. Go to the hub menu, no `[Debug] IAP` item should appear
2. Go to Settings > Experimental features
3. Enable In-app purchases
4. Go to the hub menu, a `[Debug] IAP` item should appear 
5. Tapping on `[Debug] IAP` should show a `No products available` message (for now).

### Screenshots


Hub menu | IAP Debug | Experimental Features
-|-|-
![Simulator Screen Shot - iPhone 14 Pro - 2022-10-17 at 16 23 32](https://user-images.githubusercontent.com/8739/196202719-52d9d9cb-245d-4989-9555-73705eee5a8a.png)|![Simulator Screen Shot - iPhone 14 Pro - 2022-10-17 at 16 23 34](https://user-images.githubusercontent.com/8739/196202732-845d337c-0978-4df4-8a6d-537f20739ef8.png)|![Simulator Screen Shot - iPhone 14 Pro - 2022-10-17 at 16 23 38](https://user-images.githubusercontent.com/8739/196202738-127da38e-00bf-480d-af52-70207a87893b.png)

---
